classes/category/DAO.php

```diff
@@ -42,19 +45,19 @@ class DAO extends EntityDAO
     ];
 
     /**
-     * Instantiate a new DataObject
+     * @copydoc HasParent::getParentColumn()
```

Comment:
For most editors, the `@copydoc` is not necessary. They will pick up the docblock from the trait without problems.

Reply:
Hmmm... I thought we were using those because of Doxygen or PHPDocumentor or X? -- I have forgotten... And maybe we do not use/need them any more?

Reply:
We are currently using Doxygen, but I would like to move to PHPDocumentor (and as far as I'm concerned moving gradually to its style of self-documentation would be good). As far as what IDEs tend to show, I'll defer to @NateWr's experience there -- `vim` doesn't do anything :)

Reply:
In PHPDocumentor I see: "Docblocks automatically inherit the Summary and Description of an overridden, extended or implemented element."
Thus, in child classes we do not need @copydoc.
I can't find anything about traits quickly... But... 
Both, Visual Studio Code for/with PHP and PHPDocumentor support PHPDoc, so I assume we can rely on the Visual Studio Code for/with PHP...
I will remove these @copydoc....

Reply:
hmmm... somehow I am not sure... There are tons of other copydocs in the code...
So maybe we can talk about general policy in the next dev call, so that I/we know how to proceed in such cases...
@NateWr, what does it mean that the editor "will pick up the docblock from the trait"? Does it mean that we do not need any docblock?

Reply:
> what does it mean that the editor "will pick up the docblock from the trait"? Does it mean that we do not need any docblock? Do you mean when you hover over the use of such a function it will show the summary/description? 

Yes, when a class uses a trait, it inherits those methods just like it inherit's a base class's methods. Here's an example.

Method in a trait:
```php
namespace PKP\decision\types\traits;

trait InExternalReviewRound
{
    /**
     * Get the submission file stages that are permitted to be attached to emails
     * sent in this decision
     *
     * @return array<int>
     */
    protected function getAllowedAttachmentFileStages(): array
    {
        return [
            SubmissionFile::SUBMISSION_FILE_REVIEW_ATTACHMENT,
            SubmissionFile::SUBMISSION_FILE_REVIEW_FILE,
            SubmissionFile::SUBMISSION_FILE_REVIEW_REVISION,
        ];
    }
}
```

Class uses that trait:

```php
namespace PKP\decision\types;

class Accept extends DecisionType
{
    use InExternalReviewRound;

    public function example(): void
    {
        //
    }
}
```

And this is the result of that in VS Code, and I think also PHPStorm and Sublime 2.


https://user-images.githubusercontent.com/2306629/190138554-9d0eb1b0-33b1-4d90-be97-b378a337227b.mp4

Reply:
Thanks a lot @NateWr. If that is an abstract function, do you still see the description?

Reply:
Oh, I see! You're right, the description doesn't come over from an `abstract` method. :sob: I've tried using `@copydoc` and `@inheritDoc` but I can't get the description from an `abstract` method to come over to the implemented method. 

This only seems to effect abstract methods in traits. The description from an abstract method in a parent class is coming over.

Reply:
OK. Those cases are good to know...
But, however, we can maybe see what do we want for the documentation, e.g. if we want to use PHPDocumnetor and what does that mean... If we can fully rely on what editors (e.g. VS Code) do or if we want to define how to proceed in some 'special' cases...
In this particular case, I will copy the docblock from the trait to the classes where the function is implemented... This should solve the problem for editors but surely also for documentation...